include/wil/win32_helpers.h

```diff
@@ -694,31 +729,60 @@ inline DWORD GetCurrentProcessExecutionOption(PCWSTR valueName, DWORD defaultVal
 //     missing or 0 -> don't break
 //     1 -> wait for the debugger, continue execution once it is attached
 //     2 -> wait for the debugger, break here once attached.
-inline void WaitForDebuggerPresent(bool checkRegistryConfig = true)
+namespace details
 {
-    for (;;)
+    template <typename error_policy>
+    inline void WaitForDebuggerPresent(bool checkRegistryConfig)
     {
-        auto configValue = checkRegistryConfig ? GetCurrentProcessExecutionOption(L"WaitForDebuggerPresent") : 1;
-        if (configValue == 0)
+        for (;;)
         {
-            return; // not configured, don't wait
-        }
+            DWORD configValue{1};
+            if (checkRegistryConfig)
+            {
+                DWORD registryConfigValue{};
+                // err_returncode_policy will continue running after this line on failure.  The default value of zero
+                // will apply in that case so we will still behave reasonably.
+                error_policy::HResult(details::GetCurrentProcessExecutionOptionNoThrow(L"WaitForDebuggerPresent", 0, &registryConfigValue));
+                configValue = registryConfigValue;

```

Comment:
```suggestion
                // err_returncode_policy will continue running after this line on failure.  The default value of zero
                // will apply in that case so we will still behave reasonably.
                error_policy::HResult(details::GetCurrentProcessExecutionOptionNoThrow(L"WaitForDebuggerPresent", 0, &configValue));
```
Equivalent since the first line of `details::GetCurrentProcessExecutionOptionNoThrow` assigns the default value, but with less code